src/metabase/upload.clj

```diff
@@ -461,6 +464,10 @@
   [db schema-name]
   (nil? (can-create-upload-error db schema-name)))
 
+(defn- start-timer [] (System/nanoTime))
```

Comment:
There are two reasons to use this over `System/currentTimeMillis`:

1. It's cheaper to call (doesn't require a system call)
2. It is strictly monotonic and immune to system clock resets.

Small things, but there's no reason to ever use the older method.